### DISCUSSION TITLE
Custom Edges with a button that can move along the Edge

### DISCUSSION THREAD

**Discussion by JohnCliffe** [xyflow/xyflow#2143] [Q&A]
ATM it looks like the button if added must be exactly in the middle of the edge. I've searched high and low and can't see a way to change where it is along the edge.

A good example is from here:

https://reactflow.dev/docs/examples/edges/custom-edge/

I'd love to be able to move the custom edge test a little farther down and smooth step edge further left in this example. Please let me know if this has already been asked but I can't see anything.

**Comment by adammo94** [comment]
I second that. 
What I would like to achieve is that I have a button on last curve of smoothstep or at least X number of pixels before target node

**Comment by bcakmakoglu** [comment]
As far as I am aware there's no built-in functionality to place elements along the edge besides the center.

You could implement it yourself though, [here's a very basic example](https://codesandbox.io/s/l3ue79?file=/ButtonEdge.js&from-sandpack=true).

You'll just have to figure out the x and y position of the edge with something like `ref.current.getPointAtLength(50)` (which would basically give you the x-y coordinates at length 50, as the name implies).

**Reply by ShahriarKh** [reply]
Awesome! Thank you.
